### PR TITLE
feat(database): answer Storage Health from the native backend

### PR DESCRIPTION
**Pull request by shm11C3**
## Summary

Last #2089 slice: the native DuckDB counterpart of the Storage Health family, running beside the SQLite module rather than instead of it. With this, every database consumer has a native port; nothing dispatches to it yet (that is #2134). SQLite stays authoritative.

- `native_database/storage_health.rs`: `insert_daily_records`, `refresh_daily_records`, `delete_old_data` and `latest_records`, one-to-one with the SQLite originals. Devices then records are upserted in one transaction; `refresh_daily_records` clears and re-sets `is_active` inside that same transaction, with the SQLite writer's empty-input guard kept verbatim. Conflict targets and updated column sets are reproduced exactly, including the `serial_hash = COALESCE(excluded.serial_hash, ...)` asymmetry and `first_seen_at` being absent from the update set. Every nullable real bind goes through the shared `binding::sqlite_real_binding`.
- **`COLLATE NOCASE` is not SQLite's `NOCASE`.** DuckDB has a collation by that name, but it folds Unicode case; SQLite folds only the 26 ASCII letters and then compares bytes. Measured on one list of device names, SQLite orders `Ábc, Älpha, ábc, älpha` and DuckDB's `NOCASE` orders `Ábc, ábc, Älpha, älpha`, so a disk whose OS-reported name carries a non-ASCII letter would move in the Storage Health list depending on the engine. The native query orders by a `translate` over the 26 ASCII letters instead, which the test shows agrees with SQLite. The fold direction is load-bearing: folding to upper case would sort `_` after every letter, and SQLite sorts it before `a`.
- **SQLite burns an AUTOINCREMENT id on a conflicting upsert.** `sqlite_sequence` advances even when `ON CONFLICT ... DO UPDATE` fires, so the next inserted row skips an id. The native writer calls the identity allocator unconditionally per record and burns the same ids; pinned as a delta in the identity test.
- The SQLite reader's missing-table fallback is deliberately not carried over: a database predating the storage-health migration genuinely has no records, but a finalized native database cannot be in that state, so `no such table` there means the file is not the schema it claims to be and surfaces as the engine error.
- The only edits outside the native module raise `parse_health_status`, `parse_warning_level`, `to_i64` and `from_i64` to `pub(crate)` so both backends share one definition, and `persistence::storage_health::storage_device_id` to `pub` so the differential fixture derives real `storage:hmac-sha256:v1:<digest>` identities through the production function instead of inventing ids.
- Design Doc: two paragraphs in the compatibility-boundary section recording the collation, the burned id and the missing-table decision.

## Related Issues

Completes the native ports listed in #2089 (the recurring expiry/checkpoint/reuse measurement in that issue's scope is still to do). Unblocks #2134 for this family.

## Type of Change

- [ ] Bug fix (`fix/` branch)
- [x] New feature (`feat/` branch)
- [ ] Refactoring (`refactor/` branch)
- [ ] Performance (`perf/` branch)
- [ ] Documentation (`docs/` branch)
- [ ] Dependencies update
- [ ] Other (`chore/` branch)

## Screenshots / Videos

Not applicable.

## Test Plan

- [ ] Manual testing
- [x] Unit tests

`core/tests/duckdb_storage_health.rs` (44 with the shared fixture module), differential like the other families: one fixture seeded through the production SQLite writers, converted through App migrations, the #2088 candidate builder and finalization against `get_native_schema()`, then the same question put to both engines and compared bit for bit.

- `latest_records` parity over several devices, several dates per device, ties on `MAX(date)`, inactive devices, NULL optional columns, non-ASCII and underscore device names (the collation case), and an empty result.
- Upsert parity for the same device id and the same `(device_id, date)`, every column compared afterwards, row updated in place with its id unchanged.
- `refresh_daily_records` active-flag parity with a device that disappears from the active set and one that reappears.
- Retention parity with the expected count derived from what SQLite actually deleted and a clock-relative row (no fixed-date fixture, per #2103), `storage_devices` untouched.
- Rollback: the last statement of the transaction fails and neither table changed in either engine.
- NaN and NULL for the three real columns compared on stored nullness (`typeof` on the SQLite side), not only the decoded value.
- The burned-id delta.

Local validation (macOS arm64, `--locked --offline`): fmt, `git diff --check`, clippy for both crates with and without the feature, `cargo test -p hardviz-core --features duckdb-archive -- --test-threads=1` (1,164 passed, 0 failed), the storage-health binary three times at the default parallel thread count (44/44 each), `npm run check:agent-guidance` all pass.

## Review notes

- **Test isolation is the compromise to look at.** `db::init` is process-wide, so the whole test binary shares one SQLite database while each test's native side is a private copy of the snapshot. Every cross-engine read is scoped to the ids the test seeded (bound `IN` lists on both engines), write-then-read sequences against the shared database hold one mutex because `refresh_daily_records` clears every device's active flag, and the two tests that insert brand-new rows compare rows without `id`, because SQLite's sequence has been advanced by every other test in the binary while the native copy carries the high-water mark finalization recovered. Id behaviour is pinned absolutely in the upsert test and as a delta in the identity test instead. Absolute id parity would need one SQLite database per test, which `db::init` does not allow today.
- The `COALESCE(d.display_name, s.device_id)` fallback is unreachable through the public writers on either side (the foreign key forbids an orphan record and `insert_in` refuses an empty device list), so that branch is untested in both engines.
- The rollback test relies on sqlx's default `PRAGMA foreign_keys = ON`; if that default ever changed the test would fail loudly rather than degrade.

## Checklist

- [x] Self-reviewed the code
- [x] Linting and formatting pass (`npm run lint && npm run format` / `cargo tauri-lint && cargo tauri-fmt`)
- [x] Tests pass (`npm test` / `cargo tauri-test`)
- [x] No new warnings or errors

## Summary by CodeRabbit

* **New Features**
  * Added storage health support for native DuckDB databases, including record updates, active-device refresh, retention cleanup, and latest-record retrieval.
  * Preserved compatibility with existing SQLite behavior for ordering, status values, nullable data, and transaction handling.

* **Documentation**
  * Documented native storage health compatibility and schema error behavior.

* **Tests**
  * Added comprehensive cross-database tests covering reads, writes, retention, ordering, rollback, and edge cases.

### CODE REVIEW

File: core/tests/duckdb_storage_health.rs [hunk 1]
```diff
@@ -0,0 +1,1114 @@
...
+    .to_string()
+}
+
+fn collected(offset: i64) -> String {
+  format!("{}T04:05:06Z", day(offset))
+}
+
+fn device(id: &str, display_name: &str) -> StorageDeviceRecord {
+  StorageDeviceRecord {
+    id: id.to_owned(),
```
[the remark this answers is not in the document]
Comment (shm11C3) [reply core/tests/duckdb_storage_health.rs]: Fixed in 53b2bcdb. Every fixture device id is now derived through the production function: storage_device_id was only pub(crate), so it is widened to pub with a doc comment saying why, and the test builds a SmartDiskInfo per fixture serial and hashes it with a fixture identity key, giving real storage:hmac-sha256:v1:<64 hex> values (asserted at 87 characters with the 23-character prefix). Per-test distinctness moved into the hashed input, the serial number, which is what distinguishes devices in production; no readable prefix survives into the database. serial_hash was given the production hmac-sha256:v1: shape as well, labelled as a shape rather than a real hash.

File: core/tests/duckdb_storage_health.rs [hunk 2]
```diff
@@ -0,0 +1,1114 @@
...
+#[tokio::test]
+async fn duckdb_collate_nocase_is_not_sqlites_and_the_translate_key_is() {
+  let shared = shared().await;
+  let pool = native_support::open_pool(&shared.fixture.source, false).await;
+  let oracle: Vec<String> = sqlx::query_scalar(
+    "SELECT display_name FROM storage_devices ORDER BY display_name COLLATE NOCASE",
+  )
+  .fetch_all(&pool)
+  .await
+  .unwrap();
```
[the remark this answers is not in the document]
Comment (shm11C3) [reply core/tests/duckdb_storage_health.rs]: Fixed in 53b2bcdb, and more broadly than the three queries: once ids are production hashes a LIKE prefix separates nothing, so every cross-engine read in the file (devices, records, the NaN typeof probe and all three collation queries) now builds WHERE id IN (?, ...) from the ids the test seeded, bound as parameters on both engines. The collation test also asserts the oracle row count equals the seeded set, so a scope that silently widened fails instead of passing on the wrong rows. The same reasoning exposed a second hazard the scoping alone does not fix: refresh_daily_records clears every device's active flag, so a concurrent refresh between another test's write and read-back would flip is_active in the shared SQLite database while that test's private native copy kept 1. Each write-then-read against the shared database now holds one static Mutex across both halves. Three runs at the default parallel thread count pass 44/44.
